bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/search/BahmniPatientSearchController.java

```diff
@@ -66,4 +73,19 @@ public BahmniPatientSearchController(BahmniPatientService bahmniPatientService)
             return new ResponseEntity(RestUtil.wrapErrorResponse(e, e.getMessage()), HttpStatus.BAD_REQUEST);
         }
     }
+
+    @RequestMapping(value="similar", method = RequestMethod.GET)
+    @ResponseBody
+    public ResponseEntity<AlreadyPaged<PatientResponse>> searchSimilarPerson(HttpServletRequest request,
```

Comment:
I'm not convinced that this is the right place to add this search (and that would have a big impact on the rest of the code).

See this comment: https://bahmni.atlassian.net/browse/BAH-460?focusedCommentId=12001&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-12001

This code needs to be called from the Register New Patient screen, asking the question "are there any similar patients to this one that I'm about to create?" Therefor I would expect that the client is allowed to submit an incomplete version of the patient creation object, instead of needing to construct a search that's in a different format.

If you prefer not to do it that for some reason, please give a justification.

Reply:
Hey Darius, I might be missing some context (just joined the team) but I thought that this search needs to be called as the user inputs first name / last name, and wouldn't require them to hit the submit button before checking for matches.
Did I misunderstand your comment? Do you still think we should change this?

Reply:
@djazayeri cc @mduemcke 

> Therefor I would expect that the client is allowed to submit an incomplete version of the patient creation object, instead of needing to construct a search that's in a different format

Hi Darius, I'm trying to figure this out (I've also just joined and took over from Martina). If we were to send an incomplete patient representation to the, say `BahmniPatientProfileResource` we'd still have to construct input from that representation in order to be able to search for the potentially similar patient using Lucene. As far as I understood it was suggested/agreed upon that Lucene is to be utilized for this search. Ultimately we don't want to create new patients from the lookup, but just aid the creation process, trying to avoid creation of duplicates, no?

Thus, wouldn't that mean adding a responsibility to `BahmniPatientProfileResource` which it shouldn't have? As is, the responsibilities of patient creation and searching seem more properly separated.

Though a different approach just came to mind while writing this — when creating a new patient object from the given, incomplete data, said object could obtain a new functionality `searchSimilar()` and conduct a Lucene search from there, e.g. something along `BahmniPatient::searchSimilar()`... was that what you're after? I'd probably still would want to keep that in the search controller.

Let me know what you think (or had in mind).. thanks!